homeassistant/components/renson/binary_sensor.py

```diff
@@ -0,0 +1,147 @@
+    AIR_QUALITY_CONTROL_FIELD,
+    BREEZE_ENABLE_FIELD,
+    BREEZE_MET_FIELD,
+    CO2_CONTROL_FIELD,
+    FROST_PROTECTION_FIELD,
+    HUMIDITY_CONTROL_FIELD,
+    PREHEATER_FIELD,
+    FieldEnum,
+)
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+@dataclass
+class RensonBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    field: FieldEnum
+
+
+@dataclass
+class RensonBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RensonBinarySensorEntityDescriptionMixin
+):
+    """Description of binary sensor."""
+
+
+BINARY_SENSORS: tuple[RensonBinarySensorEntityDescription, ...] = (
+    RensonBinarySensorEntityDescription(
+        name="Frost protection active",
+        key="FROST_PROTECTION_FIELD",
+        field=FROST_PROTECTION_FIELD,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_ENABLE_FIELD",
+        name="Breeze",
+        field=BREEZE_ENABLE_FIELD,
+        entity_registry_enabled_default=False,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_MET_FIELD",
+        name="Breeze conditions met",
+        field=BREEZE_MET_FIELD,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="HUMIDITY_CONTROL_FIELD",
+        name="Humidity control",
+        field=HUMIDITY_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="AIR_QUALITY_CONTROL_FIELD",
+        name="Air quality control",
+        field=AIR_QUALITY_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="CO2_CONTROL_FIELD",
+        name="CO2 control",
+        field=CO2_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="PREHEATER_FIELD",
+        name="Preheater",
+        field=PREHEATER_FIELD,
+    ),
+)
+
+
+class RensonBinarySensor(BinarySensorEntity):
+    """Get a sensor data from the Renson API and store it in the state of the class."""
+
+    def __init__(
+        self,
+        description: RensonBinarySensorEntityDescription,
+        renson_api: RensonVentilation,
+    ) -> None:
+        """Initialize class."""
+        self._state = None
+        self.renson = renson_api
+        self.field = description.field
+        self.entity_description = description
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return self._state
+
+    def update(self):
+        """Get binary data and save it in state."""
+        self._state = self.renson.get_data_boolean(self.field)
```

Comment:
The way this update method works means that each entity will now make a web request to the device to fetch all data and then pick it's field ([source](https://github.com/jimmyd-be/Renson-endura-delta-library/blob/master/renson_endura_delta/renson.py#L97-L98)). You should use the data update coordinator pattern instead.

Reply:
https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities